imageregistry/v1/types.go

```diff
@@ -7,6 +7,16 @@ import (
 	operatorv1 "github.com/openshift/api/operator/v1"
 )
 
+// ConfigMapNameReference references a config map in a specific namespace.
+// The namespace must be specified at the point of use.
+type ConfigMapNameReference struct {
+	// name is the metadata.name of the referenced config map
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Name string `json:"name"`
```

Comment:
Do you know if configmaps have particular requirements for their names?
I assume they follow one of these [conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names) right? We could add validation here to help the user and prevent them putting invalid characters in